src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/Utilities/CertUtils.cs

```diff
@@ -0,0 +1,278 @@
+// ----------------------------------------------------------------------------------
+// 
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Security.Cryptography;
+using Security.Cryptography.X509Certificates;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Class to provide methods to manage the certificates.
+    /// </summary>
+    public static class CertUtils
+    {
+        /// <summary>
+        ///     default issuer name
+        /// </summary>
+        private const string DefaultIssuer = "CN=Windows Azure Tools";
+
+        /// <summary>
+        ///     default password.
+        /// </summary>
+        private const string DefaultPassword = "";
+
+        /// <summary>
+        ///     Key size
+        /// </summary>
+        private const int KeySize2048 = 2048;
+
+        /// <summary>
+        ///     Enhancement provider
+        /// </summary>
+        private const string MsEnhancedProv = "Microsoft Enhanced Cryptographic Provider v1.0";
+
+        /// <summary>
+        ///     Client Authentication Friendly name
+        /// </summary>
+        private const string OIDClientAuthFriendlyName = "Client Authentication";
+
+        /// <summary>
+        ///     Client Authentication Value
+        /// </summary>
+        private const string OIDClientAuthValue = "1.3.6.1.5.5.7.3.2";
+
+        /// <summary>
+        ///     Method to generate a self signed certificate
+        /// </summary>
+        /// <param name="validForHours">number of hours for which the certificate is valid.</param>
+        /// <param name="subscriptionId">subscriptionId in question</param>
+        /// <param name="certificateNamePrefix">prefix for the certificate name</param>
+        /// <param name="issuer">issuer for the certificate</param>
+        /// <param name="password">certificate password</param>
+        /// <returns>certificate as an object</returns>
+        public static X509Certificate2 CreateSelfSignedCertificate(
```

Comment:
This won't work in NetStandard, correct? If I remember correctly, didn't you remove this file from your other PR?